.github/workflows/golang.yaml

```diff
@@ -0,0 +1,27 @@
+on: [push]
+
+jobs:
+  golang:
+    runs-on: ubuntu-latest # we execute everything except make in docker anyway
+    name: GoLang Basics
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Dep
+        run: make dep
+      - name: Lint
+        run: make lint
+      - name: Coverage Setup 
+        # workaround for permissions with contaner attempting to create directory
+        run: mkdir .coverage && chmod 777 .coverage
+      - name: Unit Tests
+        run: make test
+      - name: Integration Tests
+        run: make integration
+      - name: Test Coverage
+        run: make coverage
+      - name: Upload Coverage
+        uses: codecov/codecov-action@v4
+        with:
+          files: .coverage/combined.cover.out
+          token: ${{ secrets.CODECOV_TOKEN }} 
```

Comment:
question: did you have to configure this token in the project settings?

Reply:
I set it as org global. Visible only from Actions. I hope we will get rid of this very soon.

Reply:
This = codecov.